pkg/collector/corechecks/snmp/internal/devicecheck/devicecheck.go

```diff
@@ -428,3 +451,30 @@ func (d *DeviceCheck) submitPingMetrics(pingResult *pinger.Result, tags []string
 	d.sender.Gauge(pingUnreachableMetric, utils.BoolToFloat64(!pingResult.CanConnect), tags)
 	d.sender.Gauge(pingPacketLoss, pingResult.PacketLoss, tags)
 }
+
+func (d *DeviceCheck) readTagsFromCache() {
+	cacheValue, err := d.cacher.Read(d.cacheKey)
+	if err != nil {
+		log.Errorf("couldn't read cache for %s: %s", d.cacheKey, err)
+	}
+	if cacheValue == "" {
+		d.savedDynamicTags = []string{}
+	}
+	var tags []string
+	if err = json.Unmarshal([]byte(cacheValue), &tags); err != nil {
+		log.Errorf("couldn't unmarshal cache for %s: %s", d.cacheKey, err)
```

Comment:
And I think if the unmarshal fails then it should return after this rather than falling thru and setting d.savedDynamicTags.

Reply:
yes thanks !